### ISSUE TITLE
Feature request: custom word replacements

### ISSUE THREAD

**Issue by nameless-ross** [HoldOffHunger/convert-british-to-american-spellings#8]
Hi,

You're probably fed up hearing from me! The great work you have done has led me to an idea for a new feature. It would be great to customise and add additional words that may be more appropriate for different use-cases.

The best example I can think of is exchanging 'holiday' for 'vacation'. Both spellings are correct but are used differently. The ability to add custom words such as this in a class method would be really useful.

What I would propose is a new method in `AmericanBritishSpellings` which accepts an array of word replacements that would be appended to the `$word_hash` in `AmericanBritishSpellings_Words`.

I'm happy to work on a solution but I wanted to get your thoughts on this sort of functionality.

Many thanks!
Ross

**Comment by HoldOffHunger** [comment]
This is actually a pretty good idea.  There could be many reasons why someone would want the word replacement list adjusted this way or that way, without having to adjust the underlying project files.  I'll see if I can whip something up.

Feel free to keep posting any ideas or thoughts that come to mind.  For the most part, I put in issues as little reminders so I don't forget to do stuff; it's definitely not an ominous-looking todo list. =)

Thanks again, I'll see what can be done for this.

**Comment by HoldOffHunger** [comment]
I have been letting this idea roll around a bit in my head all day.

It's actually a pretty good idea.  Consider this example: A story of two people arguing about the right spelling of "labor"/"labour."  In this case, you'd want to do american/british spelling conversion based on every word but "labor", and in that case, you'd have to correct-by-hand the "labor" spellings.

So, we'd also want this option:  Not only the ability to define your own replacements, but the ability to disable replacements, i.e., `replacements=>[labor=>NULL]`, to disable `labor`/`labour` swapping, etc..  Just adding this comment so that this disabling feature is also considered.

**Comment by nameless-ross** [comment]
Yeah I agree, that would be really desirable. I've put forward a solution on my forked version:
https://github.com/nameless-ross/convert-british-to-american-spellings/commit/8d732ad2a3feb5fe0913e9fd5e80449f889ee6d1

The replacements get passed to the constructor `$args` and modifies the spellings in the `BuildSpellingReplacements()` method.

Thanks!
Ross
